engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -551,8 +551,8 @@ public void dispose() {
         renderableWorld.dispose();
         worldProvider.dispose();
         // TODO: Shift this to a better place, after a RenderGraph class has been implemented.
-        // TODO: Call ResetDefaultInstance() for every StateChange after it is a method of StateChange.
-        SetViewportToSizeOf.ResetDefaultInstance();
+        // TODO: Call disposeDefaultInstance() for every StateChange after it is a method of StateChange.
```

Comment:
That's dogmatic. 

If the user continues its way out and closes the app everything gets trashed. If the user starts a new game startup will be a smidgen faster because it doesn't have to regenerate those default instances. 

In this case it is the right thing to do. In other cases I remain unconvinced.